src/App.js

```diff
@@ -1,11 +1,18 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import Game from './pages/game'
 import { BrowserRouter,  Route, Routes } from "react-router-dom"
 import Home from './pages/home';
 
 function App() {
-  
-  const [combination, setCombination] = useState(Array.from({length: 16}).map((_, i) => i < 15 ? i + 1 : null));
+  const defaultCombination = Array.from({length: 16}).map((_, i) => i < 15 ? i + 1 : null);
+  const [combination, setCombination] = useState( () => {
+    return JSON.parse(localStorage.getItem("combination")) || defaultCombination;
+  });
```

Comment:
You'd better to move default combination generating into state creation callback. It will be a little performance improvement because of redundancy of generating new array if stored data exists.